aws/resource_aws_lambda_event_source_mapping.go

```diff
@@ -34,7 +34,7 @@ func resourceAwsLambdaEventSourceMapping() *schema.Resource {
 		Schema: map[string]*schema.Schema{
 			"event_source_arn": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
```

Comment:
I needed to make event_source_arn and self_managed_event_source both optional but one of the two needs to be set.
Is there some place where I should be validating this or should I leave it to the AWS API call failing?

Reply:
You'll want to use the `ExactlyOneOf` property.
Make `event_source_arn` and `self_managed_event_source` `Optional: true` and then add `ExactlyOneOf: []string{"event_source_arn", "self_managed_event_source"}` to them both.